### PR TITLE
[Docs] Updating frontend plugin docs [WIP]

### PR DESCRIPTION
**Pull request by joshjung**
## Hey, I just made a Pull Request!

Clarifying some things for creating plugins using both the old and new plugin systems.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/plugins/structure-of-a-plugin.md
Comment (awanlin) [inline docs/plugins/structure-of-a-plugin.md:67]: Can you revert the changes in this file, please? What was here previously was correct, this is for the current frontend system, anything from the `@backstage/frontend-plugin-api` package is intended to be for the New Frontend System 👍

File: docs/plugins/create-a-plugin.md
Comment (awanlin) [inline docs/plugins/create-a-plugin.md:42]: Would it not be better to just provide documentation for cases where you don't have a frontend component? 

The Badges plugin is one that comes to mind as it doesn't have a route at all: https://github.com/backstage/community-plugins/blob/main/workspaces/badges/plugins/badges/README.md

This is how we add it on the Demo site if you want to see a working example: https://github.com/backstage/demo/blob/master/packages/app/src/App.tsx#L54

File: docs/frontend-system/architecture/00-index.md
Comment (awanlin) [inline docs/frontend-system/architecture/00-index.md:13]: The `title` from the frontmatter at the top will be an `H1`, if we feel like there should be more than one then maybe we should break this page up?

File: docs/frontend-system/architecture/00-index.md
Comment (awanlin) [inline docs/frontend-system/architecture/00-index.md:9]: All of these notes, can we convert them to `:::info` admonitions? I think that make them more pronounced.

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute these changes @joshjung, I can't speak to the content of the New Frontend System so leaving that to the Maintainers. I did drop a couple of style comments and then for the existing frontend system I had a pair of comments as I can speak to that more confidently. 

If you want to break this up into a new and old that way it could get merged faster that would be great but that is more work for you. 👍 

Thanks again!

File: docs/plugins/structure-of-a-plugin.md
Comment (joshjung) [reply docs/plugins/structure-of-a-plugin.md:67]: This was definitely a mistake on my part :)

File: docs/plugins/create-a-plugin.md
Comment (joshjung) [reply docs/plugins/create-a-plugin.md:42]: Are you suggesting to *not* explain how it works and just show examples?

File: docs/frontend-system/architecture/00-index.md
Comment (joshjung) [reply docs/frontend-system/architecture/00-index.md:9]: Agreed

**Comment by joshjung** [comment]
Was on vacation last week, I'm back and should be able to continue work on this PR this week (time permitting here)

File: docs/plugins/create-a-plugin.md
Comment (awanlin) [reply docs/plugins/create-a-plugin.md:42]: I was thinking both but that might not have been clear. I felt like this was worth a dedicated section/paragraph explaining along with examples for this specific scenario. Does that make more sense?

Comment (joshjung) [reply docs/plugins/create-a-plugin.md:42]: Oh okay! Great, I misunderstood your use of the word "just' :) Updating now...

File: docs/frontend-system/architecture/00-index.md
Comment (joshjung) [reply docs/frontend-system/architecture/00-index.md:13]: My goal here was to simply create some structure in the sections, so that it was clear how they were related, but I can flatten it all out if you like.

File: docs/frontend-system/building-plugins/01-index.md
Comment (benjdlambert) [inline docs/frontend-system/building-plugins/01-index.md:29]: I don't think that this is true, `ExtensionDefinitions` shouldn't be exported from the package. Extensions should be wrapped up in a `Plugin` or `Module`.

Maybe we should change the above line to `Export a single FrontendFeature` or something instead?

File: docs/frontend-system/architecture/00-index.md
Comment (benjdlambert) [inline docs/frontend-system/architecture/00-index.md:32]: These are called `FrontendFeature` I believe

File: docs/frontend-system/building-plugins/01-index.md
Comment (benjdlambert) [inline docs/frontend-system/building-plugins/01-index.md:149]: Don't think we want to promote `ExtensionDefinition` type at all here, don't craft them yourself.

Use either `Blueprints` or `createExtension` directly is probably the wording we want.

File: docs/plugins/create-a-plugin.md
Comment (benjdlambert) [inline docs/plugins/create-a-plugin.md:7]: Hmm, there's no extensions really in the old system, so not sure that this is true.

File: docs/plugins/structure-of-a-plugin.md
Comment (benjdlambert) [inline docs/plugins/structure-of-a-plugin.md:61]: This is all old documentation too, so they should be using the `core-plugin-api` packages, not sure we should be surfacing the new `frontend-plugin-api` packages here.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: thanks for taking a stab at this.

Gonna pull in @Rugvip for another pair of eyes on these changes too. :pray:

File: docs/frontend-system/building-plugins/01-index.md
Comment (joshjung) [reply docs/frontend-system/building-plugins/01-index.md:29]: Your comments are really revealing on how easily it was for me to misunderstand what was going on :D One thing I might mention is that I am trying to deduce the way things work by both reading code and reading the way others have implemented the code. IIRC I deduced that you export an `ExtensionDefinition` object from an example somewhere, but I cannot recall where. I'll try to dig into this a bit today.

**Comment by joshjung** [comment]
> Hi @joshjung, just let me know when we should review this again. I'm making the assumption that you've just been busy with other things. 👍

Thanks @awanlin! Yes I have been very busy - quite a bit of traveling actually - and will get to this when I can.

**Pull request closed without merging**
